crates/oxc_syntax/src/symbol.rs

```diff
@@ -46,25 +46,26 @@ bitflags! {
         const Class                   = 1 << 5;
         const CatchVariable           = 1 << 6; // try {} catch(catch_variable) {}
         const Function                = 1 << 7;
-        const ImportBinding           = 1 << 8; // Imported ESM binding
+        const Import             = 1 << 8; // Imported ESM binding
+        const TypeImport              = 1 << 9; // Imported ESM type-only binding
```

Comment:
TypeScript doesn't have both symbols. I added it because we can quickly look up type binding and need it in #3827

Reply:
This will help with lint rules too, tyvm.